checker/src/main/java/org/checkerframework/checker/initialization/InitializationTransfer.java

```diff
@@ -151,14 +150,19 @@ protected void markInvariantFieldsAsInitialized(
     @Override
     public TransferResult<V, S> visitAssignment(AssignmentNode n, TransferInput<V, S> in) {
         TransferResult<V, S> result = super.visitAssignment(n, in);
-        assert result instanceof RegularTransferResult;
         JavaExpression lhs = JavaExpression.fromNode(n.getTarget());
 
         // If this is an assignment to a field of 'this', then mark the field as initialized.
         if (!lhs.containsUnknown()) {
             if (lhs instanceof FieldAccess) {
                 FieldAccess fa = (FieldAccess) lhs;
-                result.getRegularStore().addInitializedField(fa);
+                if (result.containsTwoStores()) {
```

Comment:
Yesterday we had discussed moving the assert here. Is handling both kinds of store necessary? Is an existing test failing without this or should we add a new test case?

Reply:
Or should it check for `!n.shouldMergeStore()` like in the super method?

Reply:
Right now only ternary expression may cause conditional transfer result, e.g. `condExpr#num0 = (obj instanceof List) `. In such cases, the LHS is never a `FieldAccess`, so we can only handle regular transfer result here.